request.go

```diff
@@ -827,22 +946,31 @@ func (r *Request) encodeRequest() {
 	}
 }
 
-func (r *Request) sendRequest() (resp *http.Response, elapsed time.Duration) {
+func (r *Request) sendRequest() (
+	resp *http.Response, conn *websocket.Conn, elapsed time.Duration,
+) {
 	if r.chain.failed() {
 		return
 	}
 
+	// TODO: print actual request made by Dialer
 	for _, printer := range r.config.Printers {
 		printer.Request(r.http)
 	}
```

Comment:
@gavv the only place I'm not sure what to do...

The problem is that `websocket.Dialer` [constructs the final HTTP request on-fly](https://github.com/gorilla/websocket/blob/v1.2.0/client.go#L199-L247) and there is no way to specify our request there. So, actually, we can't pass to `Printer` the real request (with `Upgrade` header and etc.), just preconfigured values.

The only option I see here is to repeat (copy-paste) `http.Request` construction from `gorilla/websocket`, but this will come with increased cost of maintaining (need to check changes in their lib every time).

Reply:
I thinks it's OK. Since the user can provide a custom dialer, we really don't control how a websocket request is formed, so it's not our responsibility to print it.